### ISSUE TITLE
UnhandledPromiseRejectionWarning: Error: SQLITE_ERROR: table `tmp_releases` already exist

### ISSUE THREAD

**Issue by jvlaar** [strapi/strapi#2854]
**Informations**
- **Node.js version**: v10.15.1
- **NPM version**: 6.4.1
- **Strapi version**: 3.0.0-alpha.24.1
- **Database**: SQLite (default option)
- **Operating system**: Mac OS

**What is the current behavior?**
After trying to add a field to a content type strapi crashed. Upon trying to restart it through `strapi start` the following error keeps occurring:

> ```(node:4159) UnhandledPromiseRejectionWarning: Error: SQLITE_ERROR: table `tmp_releases` already exists
>     at Client_SQLite3._query (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/dialects/sqlite3/index.js:108:12)
>     at Client_SQLite3.query (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/client.js:192:17)
>     at Runner.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/runner.js:138:36)
> From previous event:
>     at /Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/runner.js:47:21
>     at runCallback (timers.js:705:18)
>     at tryOnImmediate (timers.js:676:5)
>     at processImmediate (timers.js:658:5)
>     at process.topLevelDomainCallback (domain.js:120:23)
> From previous event:
>     at Runner.run (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/runner.js:33:30)
>     at Raw.Target.then (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/interface.js:23:43)
> (node:4159) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
> (node:4159) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
> (node:4159) UnhandledPromiseRejectionWarning: Error: SQLITE_READONLY: attempt to write a readonly database
>     at Client_SQLite3._query (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/dialects/sqlite3/index.js:108:12)
>     at Client_SQLite3.query (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/client.js:192:17)
>     at Runner.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/runner.js:138:36)
> From previous event:
>     at /Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/runner.js:47:21
> From previous event:
>     at Runner.run (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/runner.js:33:30)
>     at Builder.Target.then (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/interface.js:23:43)
>     at Child.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/bookshelf/lib/model.js:1118:36)
> From previous event:
>     at Child.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/bookshelf/lib/model.js:1117:125)
>     at runCallback (timers.js:705:18)
>     at tryOnImmediate (timers.js:676:5)
>     at processImmediate (timers.js:658:5)
>     at process.topLevelDomainCallback (domain.js:120:23)
> From previous event:
>     at Child.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/bookshelf/lib/model.js:1000:8)
> From previous event:
>     at storeTable (/Users/josvlaar/Projects/diffstrapi/node_modules/strapi-hook-bookshelf/lib/index.js:535:72)
> (node:4159) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
> (node:4159) UnhandledPromiseRejectionWarning: Error: SQLITE_READONLY: attempt to write a readonly database
>     at Client_SQLite3._query (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/dialects/sqlite3/index.js:108:12)
>     at Client_SQLite3.query (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/client.js:192:17)
>     at Runner.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/runner.js:138:36)
> From previous event:
>     at /Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/runner.js:47:21
> From previous event:
>     at Runner.run (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/runner.js:33:30)
>     at Builder.Target.then (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/interface.js:23:43)
>     at Child.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/bookshelf/lib/model.js:1118:36)
> From previous event:
>     at Child.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/bookshelf/lib/model.js:1117:125)
>     at runCallback (timers.js:705:18)
>     at tryOnImmediate (timers.js:676:5)
>     at processImmediate (timers.js:658:5)
>     at process.topLevelDomainCallback (domain.js:120:23)
> From previous event:
>     at Child.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/bookshelf/lib/model.js:1000:8)
> From previous event:
>     at storeTable (/Users/josvlaar/Projects/diffstrapi/node_modules/strapi-hook-bookshelf/lib/index.js:535:72)
> (node:4159) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 3)
> (node:4159) UnhandledPromiseRejectionWarning: Error: SQLITE_READONLY: attempt to write a readonly database
>     at Client_SQLite3._query (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/dialects/sqlite3/index.js:108:12)
>     at Client_SQLite3.query (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/client.js:192:17)
>     at Runner.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/runner.js:138:36)
> From previous event:
>     at /Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/runner.js:47:21
> From previous event:
>     at Runner.run (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/runner.js:33:30)
>     at Builder.Target.then (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/interface.js:23:43)
>     at Child.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/bookshelf/lib/model.js:1118:36)
> From previous event:
>     at Child.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/bookshelf/lib/model.js:1117:125)
>     at runCallback (timers.js:705:18)
>     at tryOnImmediate (timers.js:676:5)
>     at processImmediate (timers.js:658:5)
>     at process.topLevelDomainCallback (domain.js:120:23)
> From previous event:
>     at Child.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/bookshelf/lib/model.js:1000:8)
> From previous event:
>     at storeTable (/Users/josvlaar/Projects/diffstrapi/node_modules/strapi-hook-bookshelf/lib/index.js:535:72)
> (node:4159) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 4)
> (node:4159) UnhandledPromiseRejectionWarning: Error: SQLITE_READONLY: attempt to write a readonly database
>     at Client_SQLite3._query (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/dialects/sqlite3/index.js:108:12)
>     at Client_SQLite3.query (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/client.js:192:17)
>     at Runner.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/runner.js:138:36)
> From previous event:
>     at /Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/runner.js:47:21
> From previous event:
>     at Runner.run (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/runner.js:33:30)
>     at Builder.Target.then (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/interface.js:23:43)
>     at Child.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/bookshelf/lib/model.js:1118:36)
> From previous event:
>     at Child.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/bookshelf/lib/model.js:1117:125)
>     at runCallback (timers.js:705:18)
>     at tryOnImmediate (timers.js:676:5)
>     at processImmediate (timers.js:658:5)
>     at process.topLevelDomainCallback (domain.js:120:23)
> From previous event:
>     at Child.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/bookshelf/lib/model.js:1000:8)
> From previous event:
>     at storeTable (/Users/josvlaar/Projects/diffstrapi/node_modules/strapi-hook-bookshelf/lib/index.js:535:72)
> (node:4159) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 5)
> (node:4159) UnhandledPromiseRejectionWarning: Error: SQLITE_READONLY: attempt to write a readonly database
>     at Client_SQLite3._query (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/dialects/sqlite3/index.js:108:12)
>     at Client_SQLite3.query (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/client.js:192:17)
>     at Runner.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/runner.js:138:36)
> From previous event:
>     at /Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/runner.js:47:21
> From previous event:
>     at Runner.run (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/runner.js:33:30)
>     at Builder.Target.then (/Users/josvlaar/Projects/diffstrapi/node_modules/knex/lib/interface.js:23:43)
>     at Child.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/bookshelf/lib/model.js:1118:36)
> From previous event:
>     at Child.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/bookshelf/lib/model.js:1117:125)
>     at runCallback (timers.js:705:18)
>     at tryOnImmediate (timers.js:676:5)
>     at processImmediate (timers.js:658:5)
>     at process.topLevelDomainCallback (domain.js:120:23)
> From previous event:
>     at Child.<anonymous> (/Users/josvlaar/Projects/diffstrapi/node_modules/bookshelf/lib/model.js:1000:8)
> From previous event:
>     at storeTable (/Users/josvlaar/Projects/diffstrapi/node_modules/strapi-hook-bookshelf/lib/index.js:535:72)
> (node:4159) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 6)
> [2019-02-19T14:34:03.956Z] debug ⛔️ Server wasn't able to start properly.
> [2019-02-19T14:34:03.957Z] error (hook:bookshelf) takes too long to load
> (hook:bookshelf) takes too long to load```

**Steps to reproduce the problem**
All other fields I added previously worked fine, I'm not sure what caused this field (it was a simple string field with no extra requirements) to crash strapi.

**What is the expected behavior?**
For strapi to start

**Suggested solutions**
As looking the terminal output suggests it's something to do with the strapi-hook-bookshelf, unfortunately I don't know much more than that.

**Comment by lauriejim** [comment]
Arg if it's a random issue it will be difficult to fix.

If you remove `tmp_releases` table and restart your app does it works ?

**Comment by kasia-basia** [comment]
Hi there, I'm just playing around with strapi for the first time and got the same error after adding a new field to a previously existing and functioning model. In my case it was a Relation field. 
My config is exactly the same as jvlaar (except for Node 10.15.2). I copied the error message here: https://pastebin.com/jhSeYcXn 
So far I only figured out that strapi starts up again only after deleting the whole model folder from /api, but this is hardly a long term solution :(

**Comment by nvaken** [comment]
Same problem here. 

1. Added a new content type "Menus"
2. Added a many to many relationship "Paginas" 
3. Strapi restarted after save
4. Went to the Menus content type and noticed that it told me I couldn't add menus because of no existing fields.
5. Went to the content builder again, noticed that the relational field was still there, added a "Menu naam" string field, to try and get it to work.
6. Restarted again after hitting Save
7. Strapi crashed with

```

Warning!
We can't migrate your data due to the following error.

{ Error: SQLITE_ERROR: near "naam": syntax error
    at Client_SQLite3._query (/Users/nvaken/strapi/helloworld/node_modules/knex/lib/dialects/sqlite3/index.js:108:12)
    at Client_SQLite3.query (/Users/nvaken/strapi/helloworld/node_modules/knex/lib/client.js:192:17)
    at Runner.<anonymous> (/Users/nvaken/strapi/helloworld/node_modules/knex/lib/runner.js:138:36)
From previous event:
    at /Users/nvaken/strapi/helloworld/node_modules/knex/lib/runner.js:47:21
    at processImmediate (timers.js:632:19)
    at process.topLevelDomainCallback (domain.js:120:23)
From previous event:
    at Runner.run (/Users/nvaken/strapi/helloworld/node_modules/knex/lib/runner.js:33:30)
    at Raw.Target.then (/Users/nvaken/strapi/helloworld/node_modules/knex/lib/interface.js:23:43) errno: 1, code: 'SQLITE_ERROR' }

We created a new table "tmp_menus" with your latest changes.
We suggest you manually migrate your data from "menus" to "tmp_menus" and then to DROP and RENAME the tables.
(node:78008) UnhandledPromiseRejectionWarning: GraphQLError: Syntax Error: Expected :, found Name "naam"
    at syntaxError (/Users/nvaken/strapi/helloworld/plugins/graphql/node_modules/graphql/error/syntaxError.js:24:10)
    at expect (/Users/nvaken/strapi/helloworld/plugins/graphql/node_modules/graphql/language/parser.js:1445:32)
    at parseFieldDefinition (/Users/nvaken/strapi/helloworld/plugins/graphql/node_modules/graphql/language/parser.js:870:3)
    at many (/Users/nvaken/strapi/helloworld/plugins/graphql/node_modules/graphql/language/parser.js:1516:16)
    at parseFieldsDefinition (/Users/nvaken/strapi/helloworld/plugins/graphql/node_modules/graphql/language/parser.js:857:50)
    at parseObjectTypeDefinition (/Users/nvaken/strapi/helloworld/plugins/graphql/node_modules/graphql/language/parser.js:811:16)
    at parseTypeSystemDefinition (/Users/nvaken/strapi/helloworld/plugins/graphql/node_modules/graphql/language/parser.js:709:16)
    at parseDefinition (/Users/nvaken/strapi/helloworld/plugins/graphql/node_modules/graphql/language/parser.js:144:16)
    at many (/Users/nvaken/strapi/helloworld/plugins/graphql/node_modules/graphql/language/parser.js:1516:16)
    at parseDocument (/Users/nvaken/strapi/helloworld/plugins/graphql/node_modules/graphql/language/parser.js:115:18)
    at Object.parse (/Users/nvaken/strapi/helloworld/plugins/graphql/node_modules/graphql/language/parser.js:50:10)
    at Object.addPolymorphicUnionType (/Users/nvaken/strapi/helloworld/plugins/graphql/services/Types.js:154:8)
    at Object.generateSchema (/Users/nvaken/strapi/helloworld/plugins/graphql/services/Schema.js:179:15)
    at Function.initialize (/Users/nvaken/strapi/helloworld/plugins/graphql/hooks/graphql/index.js:148:50)
    at /usr/local/lib/node_modules/strapi/lib/hooks/index.js:19:29
    at after (/usr/local/lib/node_modules/strapi/lib/hooks/index.js:122:39)
    at /usr/local/lib/node_modules/strapi/node_modules/lodash/lodash.js:9999:23
    at Strapi.once (/usr/local/lib/node_modules/strapi/lib/hooks/index.js:133:17)
    at Object.onceWrapper (events.js:277:13)
    at Strapi.emit (events.js:189:13)
    at Strapi.EventEmitter.emit (domain.js:441:20)
    at loadedModule.initialize.call.err (/usr/local/lib/node_modules/strapi/lib/hooks/index.js:32:12)
    at Function.initialize (/Users/nvaken/strapi/helloworld/node_modules/strapi-hook-bookshelf/lib/index.js:1059:7)
(node:78008) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:78008) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
[2019-03-07T15:46:07.635Z] debug ⛔️ Server wasn't able to start properly.
[2019-03-07T15:46:07.637Z] error (hook:graphql) takes too long to load
(hook:graphql) takes too long to load
```

When I'm trying to start Strapi it instantly crashes again:

```
(node:78086) UnhandledPromiseRejectionWarning: Error: SQLITE_ERROR: table `tmp_menus` already exists
    at Client_SQLite3._query (/Users/nvaken/strapi/helloworld/node_modules/knex/lib/dialects/sqlite3/index.js:108:12)
    at Client_SQLite3.query (/Users/nvaken/strapi/helloworld/node_modules/knex/lib/client.js:192:17)
    at Runner.<anonymous> (/Users/nvaken/strapi/helloworld/node_modules/knex/lib/runner.js:138:36)
From previous event:
    at /Users/nvaken/strapi/helloworld/node_modules/knex/lib/runner.js:47:21
    at processImmediate (timers.js:632:19)
    at process.topLevelDomainCallback (domain.js:120:23)
From previous event:
    at Runner.run (/Users/nvaken/strapi/helloworld/node_modules/knex/lib/runner.js:33:30)
    at Raw.Target.then (/Users/nvaken/strapi/helloworld/node_modules/knex/lib/interface.js:23:43)
(node:78086) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:78086) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
[2019-03-07T15:47:10.443Z] debug ⛔️ Server wasn't able to start properly.
[2019-03-07T15:47:10.445Z] error (hook:bookshelf) takes too long to load
(hook:bookshelf) takes too long to load
```

**Comment by fatid** [comment]
Unfortunately, I get same error/problem. I don't know why it happens.  Now i ll try with mogodb .

**Comment by fatid** [comment]
Unfortunately, I get same error/problem. I don't know why it happens.  
Two problems occurs:
1 -I can add content ,but can not update
2- If i add new field, i have to delete tmp_xxx table.

Maybe someone need to delete table, i share the way;
- the file is in .tmp folder on root directory of strapi
- you will see data.db file 
- run this command on shell :  sqlite3 .tmp/data.db
- DROP TABLE tmp_xxx ;
- strapi start
everything works until you add new field or update content

**Comment by callumkeogan** [comment]
I'm also having the exact same problem. Even after removing the 'duplicate' table, the project simply refuses to start any more. It seems to be related to adding a relationship field to a content type.

**Comment by mehran9** [comment]
Yep, having the same issue as well.

**Comment by derrickmehaffy** [comment]
The problem here is SQLite doesnt support a standard SQL `alter table` type command without losing data.

**Comment by tony-shannon** [comment]
similar error here, after setting up new cms quickstart problem, restaurant example, then coming back to add a field.
Also went off to build API documentation and on restart got the same sqlite type error

**Comment by callumkeogan** [comment]
@lauriejim is this being looked at? It seems like it's happening to a fair few of us. It also seems like [this](https://github.com/strapi/strapi/issues/2998) might be the same problem?

**Comment by muazzami** [comment]
happening with me too.

first time trying out strapi, do a quickstart. Create few content types and voila build fails.

`(node:15452) UnhandledPromiseRejectionWarning: Error: SQLITE_ERROR: table `tmp_posts` already exists
    at Client_SQLite3._query (c:\Projects\cms\node_modules\knex\lib\dialects\sqlite3\index.js:108:12)
    at Client_SQLite3.query (c:\Projects\cms\node_modules\knex\lib\client.js:192:17)
    at Runner.<anonymous> (c:\Projects\cms\node_modules\knex\lib\runner.js:138:36)
From previous event:
    at c:\Projects\cms\node_modules\knex\lib\runner.js:47:21
    at runCallback (timers.js:705:18)
    at tryOnImmediate (timers.js:676:5)
    at processImmediate (timers.js:658:5)
    at process.topLevelDomainCallback (domain.js:120:23)
From previous event:
    at Runner.run (c:\Projects\cms\node_modules\knex\lib\runner.js:33:30)
    at Raw.Target.then (c:\Projects\cms\node_modules\knex\lib\interface.js:23:43)
(node:15452) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:15452) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
[2019-04-09T07:46:49.258Z] debug ⛔️ Server wasn't able to start properly.
[2019-04-09T07:46:49.262Z] error (hook:bookshelf) takes too long to load
(hook:bookshelf) takes too long to load`

Would like to know what causes it and how to avoid it as I was planning to use it for a project!

**Comment by gabo77** [comment]
Same here. Dropping the temp_xxx table doesn't fix it as it's being recreated automatically. In other words, the server restarts, but there's still unsaved changes in the UI, and anything I try to do with it will trigger the server restart and subsequent crash.

**Comment by brunodeangelis** [comment]
Me too. Error is: table `tmp_eventos` already exists.

Unfortunate, since I was hoping to use Strapi for an urgent project.

**Comment by derrickmehaffy** [comment]
@brunodeangelis short answer being do the use SQLite lol

**Comment by brunodeangelis** [comment]
@derrickmehaffy but I mean, is this a problem in my project config or is SQLite simply bugged?

**Comment by derrickmehaffy** [comment]
@brunodeangelis the problem is neither, the `ALTER TABLE` command in normal SQL vs SQLite are different (in that the SQLite version is far more limited) In order to modify a column in a table (and keep the data) effective what has to happen is a temporary table is made with the modifications, data is copied over from existing to the temp, the existing is deleted, and the temp is renamed like the existing one was.

This is an extremely hacky solution and the problem being is there is already an existing temp version of the table due to a previous failure (I may have the copy part flip flopped around, but either way same problem)

Strapi is alpha and SQLite support is fairly new, yes it is offered as a quick start but realistically SQLite should only be used for local development in which you don't have access to or time to setup a real SQL database (mysql or postgres). My *personal* opinion is, you should not be storing data in it that you aren't willing to lose, and you probably should not use it in production. There is no real "benefit" to using SQLite in production and in reality you have far more drawbacks to doing so. (lack of security/auth being a big one)

**Comment by derrickmehaffy** [comment]
TLDR: SQLite has no built in "ALTER TABLE / ALTER COLUMN" command and as the error suggests the previous temp table wasn't properly deleted/renamed

**Comment by qbadev** [comment]
I had same problem. But it started to happen, after I used restricted SQL word `default` as field name. Then I had to remove `tmp` and original tables using `sqlitebrowser`. After cleaning db file Strapi started to work normally.

**Comment by muazzami** [comment]
If this helps to anyone, I used separate `MySQL` as Database server `xampp` to connect and it worked fine afterwards.

**Comment by qbadev** [comment]
@muazzami do You use `sails-mysql` for connecting `strapi` to `mysql` database? If no, please share Your configuration and used packages.

**Comment by bin16** [comment]
Same problem for default sqlite db, and it occurs even when i'm not changing tables with relations. It's just there, and i would delete that table, restart server ... and it appears again and again

1. create a content type with relations // the error may be here
2. insert some data to tables
3. do a query for that table
4. stop and restart your server // boom 💥
5. remove that table and restart
6. create a new content type with relations
7. save! // The restart takes longer than expected 💥

*On my project, reason of step 7 is `tmp_products`(the content type product is created before, and modified once) too.*

```
$ strapi start
[2019-05-21T09:07:19.544Z] debug ⛔️ Server wasn't able to start properly.
[2019-05-21T09:07:19.545Z] error 
                        CREATE TABLE `tmp_products` (
                          id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,
`part_number` varchar(255)  NULL ,
`website` boolean  NULL ,
`type` varchar(255)  NULL ,
`primary` boolean  NULL ,
`secondary` boolean  NULL ,
`description` longtext  NULL ,
`productcategory` integer  NULL ,
`created_at` timestamp DEFAULT CURRENT_TIMESTAMP  NULL ,
`updated_at` timestamp DEFAULT CURRENT_TIMESTAMP  NULL 
                        )
                       - SQLITE_ERROR: table `tmp_products` already exists
{ Error: SQLITE_ERROR: table `tmp_products` already exists
    at Client_SQLite3._query (/Users/path/path/path/cms/node_modules/knex/lib/dialects/sqlite3/index.js:108:12)
    at Client_SQLite3.query (/Users/path/path/path/cms/node_modules/knex/lib/client.js:192:17)
    at Runner.<anonymous> (/Users/path/path/path/cms/node_modules/knex/lib/runner.js:138:36)
From previous event:
    at /Users/path/path/path/cms/node_modules/knex/lib/runner.js:47:21
    at processImmediate (timers.js:637:19)
    at process.topLevelDomainCallback (domain.js:120:23)
From previous event:
    at Runner.run (/Users/path/path/path/cms/node_modules/knex/lib/runner.js:33:30)
    at Raw.Target.then (/Users/path/path/path/cms/node_modules/knex/lib/interface.js:23:43) errno: 1, code: 'SQLITE_ERROR' }
```

**Comment by qbadev** [comment]
@bin16 I had similiar problem. I found out that I used reserved sql names as field names. You are trying to use primary, type etc. Check out this list [https://dev.mysql.com/doc/refman/8.0/en/keywords.html#keywords-8-0-detailed-P](url)

After I had removed reserved words, problem stopped occuring.

**Comment by bin16** [comment]
> @bin16 I had similiar problem. I found out that I used reserved sql names as field names. You are trying to use primary, type etc. Check out this list [https://dev.mysql.com/doc/refman/8.0/en/keywords.html#keywords-8-0-detailed-P](url)
> 
> After I had removed reserved words, problem stopped occuring.

Wow! Yes, I have `primary` ... Thanks for your reply! 👍

**Comment by hamedhossani** [comment]
> `tmp_releases`

I downloaded SQLiteStudio application and remove tables that i uses primary key for attribute name.
in my case **count** for product quantity.

**Comment by lauriejim** [comment]
Since the beta version (at least last version beta.15) this issue not exist anymore.

**Comment by RezaSalaari** [comment]
I had the same problem but it was solved
My problem was that I was creating the table manually with the wrong structure
In general, **this message is because your table in the database is different from your layout :)**
